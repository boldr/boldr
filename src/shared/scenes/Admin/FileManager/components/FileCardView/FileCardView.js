/* @flow */
import React from 'react';
import styled from 'styled-components';

import { Grid, Row, Col } from 'boldr-ui';
import File from '../File';

type Props = {
  selectFile: Function,
  removeMedia: Function,
  files: Array<Object>,
};

const FileCardView = (props: Props) => {
  return (
    <Grid fluid>
      <Row>
        {props.files.map(file => (
          <Col sm={12} md={4} lg={3} key={file.id}>
            <File
              file={file}
              removeMedia={props.removeMedia}
              selectFile={props.selectFile}
            />
          </Col>
        ))}
      </Row>
    </Grid>
  );
};

export default FileCardView;
