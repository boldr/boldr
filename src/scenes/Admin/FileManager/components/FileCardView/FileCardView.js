/* @flow */
import React from 'react';
// internal
import { Grid, Row, Col } from '../../../../../components/Layout';
import File from '../File';

type Props = {
  selectFile: Function,
  removeMedia: Function,
  files: Array<Object>,
};

const FileCardView = (props: Props) => {
  return (
    <Grid fluid>
      <Row>
        {props.files.map(file =>
          <Col sm={12} md={4} lg={3} key={file.id}>
            <File file={file} removeMedia={props.removeMedia} selectFile={props.selectFile} />
          </Col>,
        )}
      </Row>
    </Grid>
  );
};

export default FileCardView;
