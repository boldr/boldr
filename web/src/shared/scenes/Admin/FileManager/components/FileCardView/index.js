import FileCardView from './FileCardView';

export default FileCardView;
