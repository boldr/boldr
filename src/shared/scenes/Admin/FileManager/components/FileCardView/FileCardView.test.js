import React from 'react';
import shallow from 'react-test-renderer/shallow';
import { shallowToJson } from 'enzyme-to-json';

import fakeAttachments from '../../__fixtures__/attachments.fixture';
import FileCardView from './FileCardView';

test('<FileCardView />, renders with files as props', () => {
  const wrapper = shallow(<FileCardView files={ fakeAttachments } />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
