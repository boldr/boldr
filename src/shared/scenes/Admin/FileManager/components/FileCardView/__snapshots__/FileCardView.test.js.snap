// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileCardView />, renders with files as props 1`] = `
<Styled(Grid)
  fluid={true}
>
  <Styled(Row)>
    <Styled(Col)
      lg={3}
      md={4}
      sm={12}
    >
      <File
        file={
          Object {
            "createdAt": "2017-01-31T21:57:10.360Z",
            "fileDescription": null,
            "fileName": "BkRV_tCDe.jpg",
            "fileType": "image/jpeg",
            "id": "371de676-e800-11e6-8074-c39f798a0b4e",
            "originalName": "fullsize.jpg",
            "s3Key": null,
            "safeName": "BkRV_tCDe.jpg",
            "updatedAt": "2017-01-31T21:57:10.360Z",
            "url": "/files/BkRV_tCDe.jpg",
            "userId": "1b062e26-df71-48ce-b363-4ae9b966e7a0",
          }
        }
      />
    </Styled(Col)>
    <Styled(Col)
      lg={3}
      md={4}
      sm={12}
    >
      <File
        file={
          Object {
            "createdAt": "2017-01-31T21:57:41.632Z",
            "fileDescription": null,
            "fileName": "SkR8Ot0Pe.png",
            "fileType": "image/png",
            "id": "49c1623a-e800-11e6-a4fa-0799ce87461d",
            "originalName": "screenshot-test.png",
            "s3Key": null,
            "safeName": "SkR8Ot0Pe.png",
            "updatedAt": "2017-01-31T21:57:41.632Z",
            "url": "/files/SkR8Ot0Pe.png",
            "userId": "1b062e26-df71-48ce-b363-4ae9b966e7a0",
          }
        }
      />
    </Styled(Col)>
  </Styled(Row)>
</Styled(Grid)>
`;
